src/Common/src/System/Dynamic/Utils/ExpressionUtils.cs

```diff
@@ -171,7 +171,7 @@ public static Expression ValidateOneArgument(MethodBase method, ExpressionType n
             return arguments;
         }
 
-        public static void RequiresCanRead(Expression expression, string paramName, int idx)
+        public static void RequiresCanRead(Expression expression, string paramName, int idx = -1)
```

Comment:
I thought it was unnecessary, and dilutes the call stack especially considering exceptions are thrown here. Also, I actually prefer having the optional parameter - takes up less space, and makes the intention clear (that's just a style preference though)